### PR TITLE
fix to jamba config, asserting attention and expert offset

### PR DESCRIPTION
**Pull request by ErezSC42**
# What does this PR do?

Ensures instanced models are supported by adding assertions to attention offset and expert offset values

### CODE REVIEW

**Comment by vasqu** [comment]
I think it would be more appropriate to move this to a separate (private) function and instead of an assertion error, raise a `ValueError`.

**Comment by vasqu** [comment]
Ah and maybe add a simple test to check if the expected errors are raised 👀

File: src/transformers/models/jamba/configuration_jamba.py
Comment (vasqu) [inline src/transformers/models/jamba/configuration_jamba.py:28]: Just two small nits:
- Could you change `t_`, it's not really descriptive. 
- Maybe move it under `JambaConfig`.

**Comment by vasqu** [comment]
Jamba, cc @ArthurZucker

**Comment by ErezSC42** [comment]
I changed t_ to a be more descriptive, but since the  _check_supported_offset function is called inside JambaConfig's __init__, it cannot be a method defined inside the class

**Comment by vasqu** [comment]
Thank you! 

For `_check_supported_offset` it should be possible, see for reference:
https://github.com/huggingface/transformers/blob/f38590dade57c1f8cf8a67e9409dae8935f8c478/src/transformers/models/olmo/configuration_olmo.py#L150-L181
At least that's what I had in mind.

**Comment by ErezSC42** [comment]
my bad, it should be working now

**Comment by vasqu** [comment]
You can run `make style` which should fix the remaining issues. Thx for bearing with me :)

**Comment by ErezSC42** [comment]
Hey, all the tests passed, is this PR good for merging?

**Comment by vasqu** [comment]
It needs a review from a core maintainer, so we can just wait.

**Comment by vasqu** [comment]
Maybe cc @amyeroberts

**Review by amyeroberts — COMMENTED** [review]
LGTM - thanks for adding!

**Review by amyeroberts — APPROVED** [review]

**Pull request merged**
